examples/reshare/src/dkg/actor.rs

```diff
@@ -139,550 +169,341 @@ where
+                }
+            };
+            if !success {
+                self.failed_rounds.inc();
             }
-            Ordering::Equal => candidates,
-        }
-    }
-
-    fn choose_from_all(
-        participants: &Ordered<C::PublicKey>,
-        num_participants: u32,
-        seed: Epoch,
-    ) -> Vec<C::PublicKey> {
-        let mut rng = StdRng::seed_from_u64(seed.get());
-        participants
-            .iter()
-            .cloned()
-            .choose_multiple(&mut rng, num_participants as usize)
-    }
-
-    fn collect_all(
-        active_participants: &[C::PublicKey],
-        inactive_participants: &[C::PublicKey],
-    ) -> Ordered<C::PublicKey> {
-        active_participants
-            .iter()
-            .chain(inactive_participants.iter())
-            .cloned()
-            .collect::<Ordered<_>>()
-    }
-}
-
-#[derive(Clone)]
-struct EpochState<V: Variant> {
-    epoch: Epoch,
-    public: Option<Public<V>>,
-    share: Option<Share>,
-}
-
-impl<V: Variant> Write for EpochState<V> {
-    fn write(&self, buf: &mut impl bytes::BufMut) {
-        self.epoch.write(buf);
-        self.public.write(buf);
-        self.share.write(buf);
-    }
-}
-
-impl<V: Variant> EncodeSize for EpochState<V> {
-    fn encode_size(&self) -> usize {
-        self.epoch.encode_size() + self.public.encode_size() + self.share.encode_size()
-    }
-}
-
-impl<V: Variant> Read for EpochState<V> {
-    type Cfg = u32;
 
-    fn read_cfg(
-        buf: &mut impl bytes::Buf,
-        cfg: &Self::Cfg,
-    ) -> Result<Self, commonware_codec::Error> {
-        Ok(Self {
-            epoch: Epoch::read(buf)?,
-            public: Option::<Public<V>>::read_cfg(buf, &RangeCfg::exact(NZU32!(*cfg)))?,
-            share: Option::<Share>::read_cfg(buf, &())?,
-        })
-    }
-}
-
-#[allow(clippy::type_complexity)]
-pub(crate) struct RoundInfo<V: Variant, C: Signer> {
-    pub deal: Option<(Public<V>, Ordered<Share>, BTreeMap<u32, Ack<C::Signature>>)>,
-    pub received_shares: Vec<(C::PublicKey, Public<V>, Share)>,
-    pub local_outcome: Option<DealOutcome<C, V>>,
-    pub outcomes: Vec<DealOutcome<C, V>>,
-}
-
-impl<V: Variant, C: Signer> Default for RoundInfo<V, C> {
-    fn default() -> Self {
-        Self {
-            deal: None,
-            received_shares: Vec::new(),
-            local_outcome: None,
-            outcomes: Vec::new(),
+            info!(
+                success,
+                ?state.epoch,
+                "finalized epoch's reshare; instructing reconfiguration after reshare.",
+            );
+            let next_epoch = state.epoch.next();
+
+            // Persist the next epoch information
+            let epoch_state = EpochState {
+                epoch: next_epoch,
+                round: next_round,
+                output: next_output.clone(),
+                share: next_share.clone(),
+            };
+            self.epoch_metadata
+                .put_sync(FixedBytes::new([]), epoch_state)
```

Comment:
Is there a reason we don't keep multiple epochs of state? IMHO we should store by epoch or round number and then on startup just get the latest in epoch?

We can then delete our old share once the new epoch finalizes (to ensure we don't accidentally delete the share).